js/curation-tool.js

```diff
@@ -413,10 +464,308 @@ var vm = new Vue({
+
+    return false;
+}
+
+/**
+ * do_scnames_match_by_binomial_name(scname1, scname2)
+ *
+ * Test whether two scientific names match on the basis of their binomial names.
+ */
+function do_scnames_match_by_binomial_name(scname1, scname2) {
+    // Step 1. Try matching by explicit binomial name, if one is available.
+    if(scname1.hasOwnProperty('binomialName') && scname2.hasOwnProperty('binomialName')) {
+        if(scname1.binomialName.trim() !== '' && scname1.binomialName.toLowerCase().trim() === scname2.binomialName.toLowerCase().trim())
+            return true;
+    }
+
+    // Step 2. Otherwise, try to extract the binomial name from the scientificName
+    // and compare those.
+    let binomial1 = vm.get_binomial_name(scname1);
+    let binomial2 = vm.get_binomial_name(scname2);
+
+    if(binomial1 !== undefined && binomial2 !== undefined && binomial1.trim() !== '' && binomial1.trim() === binomial2.trim())
+        return true;
+
+    return false;
+}
+
+/**
+ * do_tunits_match_by_binomial_name(tunit1, tunit2)
+ *
+ * Test whether two Tunits match on the basis of their binomial names.
+ */
+function do_tunits_match_by_binomial_name(tunit1, tunit2) {
+    //
+
+    if(tunit1 === undefined || tunit2 === undefined) return false;
+    if(tunit1.hasOwnProperty('scientificNames') && tunit2.hasOwnProperty('scientificNames')) {
+        // Each external reference is a URL as a string.
+        for(let scname1 of tunit1.scientificNames) {
+            for(let scname2 of tunit2.scientificNames) {
+                if(do_scnames_match_by_binomial_name(scname1, scname2))
+                    return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+/**
+ * do_tunits_match_by_specimen_identifier(tunit1, tunit2)
+ *
+ * Test whether two Tunits match on the basis of their specimen identifiers.
+ */
+function do_tunits_match_by_specimen_identifier(tunit1, tunit2) {
+    if(tunit1 === undefined || tunit2 === undefined) return false;
+    if(tunit1.hasOwnProperty('includesSpecimens') && tunit2.hasOwnProperty('includesSpecimens')) {
+        // Convert specimen identifiers (if present) into a standard format and compare those.
+        for(let specimen1 of tunit1.includesSpecimens) {
+            for(let specimen2 of tunit2.includesSpecimens) {
+                if(
+                    get_specimen_identifier_as_urn(specimen1) !== undefined &&
+                    get_specimen_identifier_as_urn(specimen2) !== undefined &&
+                    get_specimen_identifier_as_urn(specimen1) === get_specimen_identifier_as_urn(specimen2)
+                )
+                    return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+/**
+ * get_specimen_identifier_as_urn(specimen)
+ *
+ * Given a specimen, return a specimen identifier as a URN in the form:
+ *   "urn:catalog:" + institutionCode (if present) + ':' + collectionCode (if present) + ':' + catalogNumber (if present)
+ */
+function get_specimen_identifier_as_urn(specimen) {
```

Comment:
See above.

Reply:
Issue #35 covers the Specimen class as well. I think this name is okay for now, but please let me know if you have a suggestion for a better name! When I convert it into camelCase, I plan to rename it to `getSpecimenIdentifierAsURN()`, which should be a little clearer.